pandas/core/dtypes/concat.py

```diff
@@ -45,6 +46,8 @@ def get_dtype_kinds(l):
             # if to_concat contains different tz,
             # the result must be object dtype
             typ = str(arr.dtype)
+        elif is_range(arr):
```

Comment:
Yep, if it is for a single call, I would just do the `isinstance(arr, ABCRangeIndex)` here instead of defining the new function